name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/JTDirektbankPDFExtractor.java

```diff
@@ -0,0 +1,152 @@
+        type.addBlock(removalBlock);
+
+        Block interestBlock = new Block(INTEREST_2023);
+        interestBlock.set(interestTransaction(type, INTEREST_2023));
+        type.addBlock(interestBlock);
+    }
+
+
+    private Transaction<AccountTransaction> depositTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DEPOSIT);
+            return entry;})
+                        .section("date", "amount")
+                        .match(regex)
+                        .assign(assignmentsProvider(type))
+                        .wrap(TransactionItem::new);
+    }
+
+    private Transaction<AccountTransaction> interestTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.INTEREST);
+            return entry;})
+                        .section("date", "amount")
+                        .match(regex)
+                        .assign(assignmentsProvider(type))
+                        .wrap(TransactionItem::new);
+    }
+
+    private Transaction<AccountTransaction> removalTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.REMOVAL);
+            return entry;})
+                        .section("date", "amount")
+                        .match(regex)
+                        .assign(assignmentsProvider(type))
+                        .wrap(TransactionItem::new);
+    }
+
+    private BiConsumer<AccountTransaction, ParsedData> assignmentsProvider(DocumentType type)
+    {
+        return (transaction, matcherMap) -> {
+            Map<String, String> context = type.getCurrentContext();
+
+            boolean transactionsHaveFullDate = CONTEXT_VALUE_TRUE.equals(context.get(CONTEXT_KEY_TRANSACTIONS_HAVE_FULL_DATE));
+
+            String date = matcherMap.get("date");
+
+            if (!transactionsHaveFullDate)
+            {
+                date += context.get(CONTEXT_KEY_YEAR);
+            }
+
+            transaction.setDateTime(asDate(date));
+            transaction.setAmount(asAmount(matcherMap.get("amount")));
+            transaction.setCurrencyCode(context.get(JTDirektbankPDFExtractor.CONTEXT_KEY_CURRENCY));
+        };
+    }
+
+    private BiConsumer<DocumentContext, String[]> contextProvider2023()
+    {
+        return (context, lines) -> {
+            Pattern yearPattern = Pattern.compile(".*Kontoauszug Nr\\.[\\s]{1,}[\\d+]\\/(?<year>[\\d]{4})");
+            Pattern currencyPattern = Pattern
+                            .compile("(?<currency>[\\w]{3})\\-Konto Kontonummer.*");
```

Comment:
Sorry, my fault. Here is the correct regex. 👍🏻 
```suggestion
            Pattern yearPattern = Pattern.compile("^.*Kontoauszug Nr\\.[\\s]{1,}[\\d]{1,2}\/(?<year>[\\d]{4})$");
            Pattern currencyPattern = Pattern.compile("^(?<currency>[\w]{3})\-Konto Kontonummer.*$");
```